src/page/FindPassword/index.tsx

```diff
@@ -0,0 +1,163 @@
+import { SHA256 } from 'crypto-js';
+import { useSnackBar } from 'ts/useSnackBar';
+import { AxiosError } from 'axios';
+import { useNavigate } from 'react-router-dom';
+import * as S from './style';
+
+const initialState = {
+  email: '',
+  token: '',
+  password: '',
+  passwordCheck: '',
+};
+
+interface Step {
+  email: boolean,
+  token: boolean,
+  password: boolean,
+}
+
+const initialStep = {
+  email: true,
+  token: false,
+  password: false,
+};
+
+export default function FindPassword() {
+  const [passwordChangeInfo, setPasswordChangeInfo] = useState<ChangePasswordForm>(initialState);
+  const [step, setStep] = useState<Step>(initialStep);
+  const openSnackBar = useSnackBar();
+  const navigate = useNavigate();
+
+  const handleEmailInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPasswordChangeInfo({ ...passwordChangeInfo, email: e.target.value });
+  };
+
+  const handleTokenInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPasswordChangeInfo({ ...passwordChangeInfo, token: e.target.value });
+  };
+
+  const handlePasswordInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPasswordChangeInfo({ ...passwordChangeInfo, password: e.target.value });
+  };
+
+  const handlePasswordCheckInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPasswordChangeInfo({ ...passwordChangeInfo, passwordCheck: e.target.value });
+  };
+
+  const requestEamil = async (email: RequestChangePassword) => {
+    try {
+      await requestChangePassword(email);
+      setStep({ ...step, token: true });
+    } catch (e) {
+      if (e instanceof AxiosError) openSnackBar({ type: 'error', message: e.response?.data.message });
+    }
+  };
+
+  const certification = async (info: CertificationToken) => {
+    try {
+      await certificationToken(info);
+      setStep({ ...step, password: true });
+    } catch (e) {
+      if (e instanceof AxiosError) openSnackBar({ type: 'error', message: e.response?.data.message });
+    }
+  };
+
+  const change = async (info: ChangePassword) => {
+    try {
+      if (info.password === passwordChangeInfo.passwordCheck) {
+        await changePassword({
+          email: info.email,
+          token: info.token,
+          password: SHA256(info.password).toString(),
+        });
+        openSnackBar({ type: 'success', message: '비밀번호를 성공적으로 변경했습니다.' });
+        navigate('/');
+      } else {
+        openSnackBar({ type: 'error', message: '비밀번호가 일치하지 않습니다.' });
+      }
+    } catch (e) {
+      if (e instanceof AxiosError) openSnackBar({ type: 'error', message: e.response?.data.message });
+    }
+  };
+
+  const changePasswordStep = () => {
+    if (step.password) {
+      change({
+        email: passwordChangeInfo.email,
+        token: passwordChangeInfo.token,
+        password: passwordChangeInfo.password,
+      });
+    } else if (step.token) {
+      certification({
+        email: passwordChangeInfo.email,
+        token: passwordChangeInfo.token,
+      });
```

Comment:
else if가 아닌 그냥 if를 사용해도 똑같이 동작할 것 같아요
```suggestion
    }
    if (step.token) {
      certification({
        email: passwordChangeInfo.email,
        token: passwordChangeInfo.token,
      });
```

Reply:
그냥 if문을 적용하면
마지막 비밀번호를 바꾸는 호출을  요청했을 때
인증번호를 확인하는 certification이 한 번 더 호출돼서 else if문을 적용했습니다